third_party/terraform/resources/resource_bigquery_table.go

```diff
@@ -683,6 +694,26 @@ func resourceBigQueryTableRead(d *schema.ResourceData, meta interface{}) error {
 			return err
 		}
 
+		edc := map[string]interface{}(nil)
+
+		if v, ok := d.GetOk("external_data_configuration"); ok {
+			// The API response doesn't return the `external_data_configuration.schema`
+			// used when creating the table and it cannot be queried.
+			// After creation, a computed schema is stored in the toplevel `schema`,
+			// which combines `external_data_configuration.schema`
+			// with any hive partioning fields found in the `source_uri_prefix`.
+			// So just assume the configured schema has been applied after successful
+			// creation, by copying the configured value back into the resource schema.
+			// This avoids that reading back this field will be identified as a change.
+			// The `ForceNew=true` on `external_data_configuration.schema` will ensure
```

Comment:
I don't totally follow why not reading back the external schema requires us to do ForceNew. Couldn't we still allow updating that field even while not detecting drift on it?

Reply:
The ForceNew is not required in terms of reading back the data, I just consider this better UX wrt the expectations of the user, e.g. if you change the schema, the most probable course I think is to recreate the table with that schema. Afaik you cannot change the schema of an existing table in place. 

The `external_data_configuration.schema` is only used as an _input parameter_ for creating the table, when we read back the table, this field is always _empty_. There is a computed `schema` returned on top level, which reflects the effective schema of the created table, however this value is calculated by combining the schema provided here and any other field/type mappings it can infer by autodetection and/or inferred from the `source_uri_prefix`. I wanted to avoid having to determine if `external_data_configuration.schema` is accurately reflected in the computed schema and reimplement BQ's logic in doing so, so I just assume after creation this is successfully reflected, hence I just ignore this field by making sure there are no changes with what is configured.

Perhaps there's a smarter way to do this?

Reply:
That makes sense. If it's not clear that it can be updated and users wouldn't necessarily expect it to be, we can leave it as `ForceNew`. Worst thing that happens is someone files an issue to ask for update support.